### PR TITLE
INSP & ANN: Disable analysis on cfg-disabled code

### PR DESCRIPTION
**Pull request by ortem**
* Disable inspections on cfg-disabled code via `RsProblemsHolder` proxy
* Disable annotations on cfg-disabled code via `RsAnnotationHolder` proxy
* Disable parameter and type hints on cfg-disabled code
* Gray cfg-disabled code via `RsCfgDisabledCodeAnnotator`

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/utils/CfgUtils.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+package org.rust.ide.utils
+
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.ext.RsDocAndAttributeOwner
+import org.rust.lang.core.psi.ext.ancestors
+import org.rust.lang.core.psi.ext.isEnabledByCfg
+
+/**
+ * Element is conditionally enabled iff it is not conditionally disabled by any of parent [RsDocAndAttributeOwner]
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/utils/CfgUtils.kt]: type in `iff`

Comment (ortem) [reply src/main/kotlin/org/rust/ide/utils/CfgUtils.kt]: https://en.wikipedia.org/wiki/If_and_only_if

**Comment by vlad20012** [comment]
Add color for the dark theme

**Pull request merged**